tests/test_parallel_corpus.py

```diff
@@ -0,0 +1,155 @@
+            'This is a sample text with a mojibake dash â€”\t'
+            'Este es un texto de muestra con un guión mojibake â€”',
+            'This is a sample text with a mojibake dash —\t'
+            'Este es un texto de muestra con un guión mojibake —',
+            'en', 'es'
+        ),
+        pytest.param(
+            '♪ I am happy with these notes by my sides. ♪\t'
+            '♪ Estoy feliz con estas notas a mi lado. ♪',
+            '♪ I am happy with these notes by my sides. ♪\t'
+            '♪ Estoy feliz con estas notas a mi lado. ♪',
+            'en', 'es'
+        ),
+        pytest.param(
+            '♪ Sadly, I will be missing a note on the other side. ♪\t'
+            '♪ Lamentablemente, seré la misión una nota en el otro lado.',
+            '',
+            'en', 'es'
+        ),
+        pytest.param(
+            'Short text.\tछोटो पाठ',
+            'Short text.\tछोटो पाठ',
+            'en', 'es'
+        ),
+        pytest.param(
+            'This is a sample text with correct language on both sides.\t'
+            'Este es un texto de muestra con el lenguaje correcto en ambos lados.',
+            'This is a sample text with correct language on both sides.\t'
+            'Este es un texto de muestra con el lenguaje correcto en ambos lados.',
+            'en', 'es'
+        ),
+        pytest.param(
+            'This is a sample text with wrong language on one side.\t'
+            'これは、一方の言語が間違っているサンプルテキストです。',
+            '',
+            'en', 'es'
+        ),
+
+    ]
+)
+def test_cleanup_parallel_file(tmpdir, text, expected, lang1, lang2, path_to_ft_model):
+    in_file = [text]
+    out_file = StringIO()
+
+    cleanup_parallel_file(
+        in_file, out_file, path_to_ft_model, lang1, lang2
+    )
+    out_file.seek(0)
+    assert out_file.read() == expected
+
+
+@pytest.fixture(scope="session")
+def path_to_sp_model(tmpdir_factory):
+    input_dir = tmpdir_factory.mktemp('input')
+    input_file = input_dir.join('test_input.txt')
+    input_file.write(
+        'abcdefghijklmnopqrstuvwxyz\n'
+        'ABCDEFGHIJKLMNOPQRSTUVWXYZ\n'
+    )
+
+    model_dir = tmpdir_factory.mktemp('model')
+    model_prefix = str(model_dir.join('m'))
+
+    lang = 'en'
+    train_sentencepiece(input_file, model_prefix, lang)
+    return model_prefix + '.model'
+
+
+def test_train_sentencepiece(path_to_sp_model):
+    sp = sentencepiece.SentencePieceProcessor()
+
+    # Check if the model exists and we can load it
+    assert os.path.exists(path_to_sp_model)
+    assert sp.Load(path_to_sp_model)
+
+
+def test_get_vocabulary_from_sp(path_to_sp_model):
+    out_file = StringIO()
+    get_vocabulary_from_sp(out_file, path_to_sp_model)
+
+    # Check if we get correct number of vocabulary
+    # Should be 26 lower case alphabets + '▁'
+    out_file.seek(0)
+    count = len(out_file.readlines())
+    assert count == 27
+
+
+@pytest.mark.parametrize(
+    'text, expected',
+    [
+        (DECODED[0], ENCODED[0]),
```

Comment:
Can all of this be replaced with `zip(DECODED, ENCODED)`?

Reply:
Yes, it can be.